File: lib/src/models/payment.dart
import 'package:sneakerx/src/models/enums/payment_method.dart';
import 'package:sneakerx/src/models/enums/payment_status.dart';

class Payment {
  final int paymentId;
  final int orderId;
  final int userId;
  final double amount;
  final SneakerXPaymentMethod paymentMethod;
  final PaymentStatus paymentStatus;
  final String transactionId;
  final DateTime createdAt;

  Payment({
    required this.paymentId,
    required this.orderId,
    required this.userId,
    required this.amount,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.transactionId,
    required this.createdAt
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      paymentId: json['paymentId'],
      orderId: json['orderId'],
      userId: json['userId'],
      amount: json['amount'],
      paymentMethod: SneakerXPaymentMethod.values.byName(json['paymentMethod']),
      paymentStatus: PaymentStatus.values.byName(json['paymentStatus']),
      transactionId: json['transactionId'],
      createdAt: DateTime.parse(json['createdAt'])
    );
  }
}

File: src/main/java/com/example/sneakerx/entities/Payment.java
package com.example.sneakerx.entities;

import com.example.sneakerx.entities.enums.PaymentMethod;
import com.example.sneakerx.entities.enums.PaymentStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "payments")
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer paymentId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Order order;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private User user;

    private Double amount;

    @Enumerated(EnumType.STRING)
    private PaymentMethod paymentMethod;

    @Enumerated(EnumType.STRING)
    private PaymentStatus paymentStatus;

    private String transactionId;
    private LocalDateTime paidAt;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
